pkg/merger/merger.go

```diff
@@ -0,0 +1,129 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package merger
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/sirupsen/logrus"
+	"gopkg.in/yaml.v2"
+
+	"github.com/GoogleCloudPlatform/testgrid/config"
+	configpb "github.com/GoogleCloudPlatform/testgrid/pb/config"
+	"github.com/GoogleCloudPlatform/testgrid/util/gcs"
+)
+
+type configList struct {
+	Target  string    `json:"Target"`
+	Sources []sources `json:"Sources"`
+}
+
+type sources struct {
+	Name     string `json:"Name"`
+	Location string `json:"Location"`
+}
+
+// ParseAndCheck parses and checks the configuration file for common errors
+func ParseAndCheck(data []byte) (target *gcs.Path, sources map[string]*gcs.Path, err error) {
+	cfg := &configList{}
+	err = yaml.Unmarshal(data, cfg)
+	if err != nil {
+		return
+	}
+
+	target, err = gcs.NewPath(cfg.Target)
+	if err != nil {
+		return
+	}
+
+	sources = make(map[string]*gcs.Path)
+	for _, source := range cfg.Sources {
+		if _, exists := sources[source.Name]; exists {
+			return nil, nil, fmt.Errorf("duplicated name %s", source.Name)
+		}
+		path, err := gcs.NewPath(source.Location)
+		if err != nil {
+			return nil, nil, err
+		}
+		sources[source.Name] = path
+	}
+
+	if len(sources) == 0 {
```

Comment:
Seems like we can do this earlier by checking `len(cfg.Sources) == 0`, right?

Reply:
Done